src/experimental/information_estimates.cpp

```diff
@@ -0,0 +1,525 @@
+#include <math.h>
+#include <cmath>
+#include "slam_toolbox/experimental/information_estimates.hpp"
+
+#include <iostream>
+
+InformationEstimates::InformationEstimates(kt_double sensor_range, kt_double resolution, kt_double lambda, kt_double nu)
+    : m_max_sensor_range{sensor_range},
+    m_cell_resol{resolution},
+    m_obs_lambda{lambda},
+    m_obs_nu{nu}
+{
+}
+
+InformationEstimates::InformationEstimates()
+    : m_max_sensor_range{25.0},
+    m_cell_resol{0.1},
+    m_obs_lambda{0.35},
+    m_obs_nu{0.28}
+{
+}
+
+
+void InformationEstimates::resizeGridFromScans(std::vector<karto::LocalizedRangeScan *> const & range_scans)
+{
+    m_lower_limit_x = range_scans[0]->GetCorrectedPose().GetX();
+    m_lower_limit_y = range_scans[0]->GetCorrectedPose().GetY();
+    m_upper_limit_x = range_scans[0]->GetCorrectedPose().GetX();
+    m_upper_limit_y = range_scans[0]->GetCorrectedPose().GetY();
+
+    for (const auto &scan : range_scans)
+    {
+        if (scan == nullptr)
+        {
+            continue;
+        }
+
+        karto::Pose2 pose = scan->GetCorrectedPose();
+        kt_double minimumAngle = scan->GetLaserRangeFinder()->GetMinimumAngle();
+        kt_double angularResolution = scan->GetLaserRangeFinder()->GetAngularResolution();
+
+        // Finding the lower limit pose
+        m_lower_limit_x = std::min(pose.GetX(), m_lower_limit_x);
+        m_lower_limit_y = std::min(pose.GetY(), m_lower_limit_y);
+
+        // Finding the higher limit pose
+        m_upper_limit_x = std::max(pose.GetX(), m_upper_limit_x);
+        m_upper_limit_y = std::max(pose.GetY(), m_upper_limit_y);
+    }
+
+    // Map dimensions
+    m_map_dim_x = std::fabs(m_upper_limit_x - m_lower_limit_x) + (2 * m_max_sensor_range);
+    m_map_dim_y = std::fabs(m_upper_limit_y - m_lower_limit_y) + (2 * m_max_sensor_range);
+
+    // Get the number of cells
+    int n_cells_x = static_cast<int>(m_map_dim_x / m_cell_resol);
+    int n_cells_y = static_cast<int>(m_map_dim_y / m_cell_resol);
+
+    // Resize the grid with new number of cells
+    m_mutual_grid.resize(n_cells_x, n_cells_y);
+    m_visited_grid.resize(n_cells_x, n_cells_y);
+}
+
+
+int InformationEstimates::findClosestLaserIndexToCell(
+    kt_bool & skip_cell_eval,
+    kt_double const & angle_to_cell,
+    kt_double const & scan_pose_heading,
+    karto::LaserRangeFinder *laser_range_finder)
+{
+    kt_double angle_between = angle_to_cell - scan_pose_heading;
+    kt_double angle_diff = atan2(sin(angle_between), cos(angle_between));
```

Comment:
To be completely honest, this was the only solution that came up to my mind.

I will dig into it, but if you have any suggestion for this, it would be so valuable.

Reply:
> but if you have any suggestion for this, it would be so valuable.

@kmilo7204 not really, just curious.